### PR TITLE
Mz post qs

### PR DESCRIPTION
**Pull request by MarZmu**
added post for answers/questions, that will also accept reviews once the controller is set up for handling review info

### CODE REVIEW

File: client/src/components/QModal.jsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+    console.log(document.getElementById('newQA').value);
+    console.log(document.getElementById('nick').value);
+    console.log(document.getElementById('email').value);
+
+    return postQ({
+      body: document.getElementById('newQA').value,
+      name: document.getElementById('nick').value,
+      email: document.getElementById('email').value,
+      product_id: pID
     });
```
Comment (Dalphus) [inline client/src/components/QModal.jsx:20]: please use `form`'s `onSubmit` attribute to submit form data.

don't use `document.getElementById`, instead save form element values in state using props, or reference values in the submit event object passed to `onSubmit`

File: server/controllers.js [hunk 2]
```diff
@@ -56,3 +56,26 @@ module.exports.put = (endpoint, config) => {
...
+    url = new URL(join(api, route[0], body.product_id, route[1])).href;
+  }
+
+  const options = { headers: { Authorization: process.env.GITHUB_API_KEY } };
+  console.log(body);
+  // return a promise for the requested object
+  return axios.post(url, body, options)
+    .then((res) => res.body)
+    .catch((err) => {
+      console.log(err.toJSON());
```
Comment (Dalphus) [inline server/controllers.js:79]: use `console.error` to log errors

**Review by Dalphus — CHANGES_REQUESTED** [review]
Your code also has several linter errors that need resolving
Also, please enable the `max-len` eslint rule, as some of your lines too long to easily read

File: server/controllers.js (hunk 2 above)
Comment (MarZmu) [reply server/controllers.js:79]: this logs the actual error vs using console error new Error. it implements the built in function of express

File: client/src/components/QModal.jsx [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
   function onSubmitClickVal() {
-    return submitForm({
-      body: document.getElementById('#newQA'),
-      name: document.getElementById('#nick'),
-      email: document.getElementById('#email'),
-      product_id: product.product_id
+
+    console.log(document.getElementById('newQA').value);
+    console.log(document.getElementById('nick').value);
+    console.log(document.getElementById('email').value);
```
[the remark this answers is not in the document]
Comment (MarZmu) [reply client/src/components/QModal.jsx:13]: updated

File: client/src/components/QModal.jsx [hunk 4]
```diff
@@ -1,16 +1,22 @@
 import React from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch } from 'react-redux';
+import { postQ } from './QAhelp/QArequests';
 
-function QModal({ product }) {
+function QModal({ pID }) {
   // const qModal = useSelector((state) => state.qList.mo)
   const dispatch = useDispatch();
```
[the remark this answers is not in the document]
Comment (MarZmu) [reply client/src/components/QModal.jsx:7]: utilizing useDispatch hooks to simplify codebase. Easier for use with the component

File: client/src/components/QModal.jsx (hunk 1 above)
Comment (MarZmu) [reply client/src/components/QModal.jsx:20]: updated

File: server/controllers.js (hunk 2 above)
Comment (Dalphus) [reply server/controllers.js:78]: console.error highlights the log in red on the console, the change I am suggesting is to use:
```javascript
console.error(err.toJSON());
```

**Review by Dalphus — APPROVED** [review]

**Pull request merged**